packages/legacy/src/queries/tree.ts

```diff
@@ -128,3 +142,32 @@ export const useInventoryTreeQuery = <T extends InventoryTree>(
     apiMockData as T
   );
 };
+
+function createTreeFromVMs(vms: ICommonTreeObject[]): InventoryTree {
+  // Create children nodes from VMs
+  const children: InventoryTree[] = vms.map((vm) => ({
+    kind: 'VM',
+    object: { id: vm.name, ...vm },
```

Comment:
cc:// @bkhizgiy - hi, the VM does not have `id` or `uid` falling back to `name` if you add id, we can drop this fallback mechanizm